Meta/Lagom/Tools/CodeGenerators/LibWeb/BindingsGenerator/IDLGenerators.cpp

```diff
@@ -1487,11 +1487,19 @@ static void generate_wrap_statement(SourceGenerator& generator, DeprecatedString
 
     if (type.is_nullable() && !is<UnionType>(type)) {
         if (type.is_string()) {
-            scoped_generator.append(R"~~~(
+            if (!interface.extended_attributes.contains("UseNewAKString")) {
+                scoped_generator.append(R"~~~(
     if (@value@.is_null()) {
         @result_expression@ JS::js_null();
     } else {
 )~~~");
+            } else {
+                scoped_generator.append(R"~~~(
+    if (@value@.is_empty()) {
+        @result_expression@ JS::js_null();
```

Comment:
That looks extremly wrong - these functions should return an `Optional<String>`, as a replacement for the `DeprecatedString` null state.

Reply:
Absolutely, should have known better. Fixed